binding.gyp

```diff
@@ -1,50 +1,46 @@
 {
+  "variables": {
+    "realm_node_build_as_library%": "0"
+  },
   "includes": [
     "src/node/gyp/target_defaults.gypi",
     "src/node/gyp/realm.gyp"
   ],
   "targets": [
     {
-      "variables": {
-        "developer_edition%": "0"
-      },
       "target_name": "realm",
       "dependencies": [
         "object-store"
       ],
       "sources": [
-        "src/node/node_sync_logger.cpp",
-        "src/node/node_init.cpp",
         "src/node/platform.cpp",
         "src/js_realm.cpp"
       ],
       "include_dirs": [
         "src"
       ],
-      "defines": [ "REALM_DEVELOPER_EDITION=<(developer_edition)" ],
-      "link_settings": {
-        "ldflags": [
-          "-Wl,--exclude-libs=ALL"
-        ]
-      },
-      "xcode_settings": {
-        "OTHER_LDFLAGS": [ "-Xlinker -unexported_symbol -Xlinker '*'" ]
-      }
+      "conditions": [
+        ["realm_node_build_as_library", {
+          "type": "static_library",
+          "export_dependent_settings": [ "object-store" ]
+        }, {
+          "sources": [
+            "src/node/node_init.cpp"
+          ]
+        }]
+      ]
     },
     {
-      "variables": {
-        "object-store-include-dirs": [
-          "src/object-store/src",
-          "src/object-store/src/impl",
-          "src/object-store/src/impl/apple",
-          "src/object-store/src/parser",
-          "src/object-store/external/pegtl"
-        ]
-      },
       "target_name": "object-store",
-      "dependencies": [ "realm-sync" ], # sync also includes core
+      "dependencies": [ "realm-core" ],
       "type": "static_library",
-      "include_dirs": [ "<@(object-store-include-dirs)" ],
```

Comment:
You'll notice I list the include directories twice - once here and once in `all_dependent_settings`. I used to use a variable and reference that, but gyp doesn't resolve the absolute paths of elements in list variables when expanding them in the downstream target via `all_dependent_settings` - instead of using the `object-store` target as a base directory, it uses the importing target, which results in wrong absolute paths.